### PR TITLE
Render full instruction pages at build time

### LINKED ISSUE #261: Improve search engine findability for help pages
The current one-page design is nice and fast, but it's hard for search engines to find. For instance, the instructions to install on Ubuntu are at https://certbot.eff.org/#ubuntutyakkety-apache. Because the locator for that specific page is based on a URL fragment instead of the whole URL, search engines see it as identical to https://certbot.eff.org/.

We should break out each platform/software combination into its own page, with a full URL, like https://certbot.eff.org/lets-encrypt-ubuntu-yakkety-apache. Note that the # is removed, and the keywords people are likely to search for are separated out with hyphens. We should also ensure each page has a title that reflects what the page is about, e.g. "How to get a Let's Encrypt certificate on Ubuntu with Certbot and Apache," and a 1-sentence blurb at the top to give context to people arriving directly from a search rather than from the main page.

### PR DESCRIPTION
**Pull request by vbrown608**
Renders the instruction page for each (os, webserver) pair at build time. This lets us create standalone pages that are accessible to users with javascript disabled. It also moves more rendering and routing into jekyll/nginx, which I think is preferable to the very custom javascript.

When users with javascript enabled submit the widget form, I'm using js to update the url + page history to the new path. For users without javascript, the form submits to something like certbot.eff.org/?server=apache&os=arch. Nginx parses the query string and rewrites it to the correct url.

Individual instruction pages are now found at, eg, certbot.eff.org/i/arch-apache. Do we want to tweak that to include a keyword about letsencrypt? We could do certbot.eff.org/letsencrypt-instructions/arch-apache, for example.

We'll need to update the nginx config before we deploy this. I'm working with techops on that.

Resolves #261, #294 and #111

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,9 +4,17 @@ Website for [EFF's Certbot](https://certbot.eff.org/) project. Uses Jekyll for s
 
 [![Build Status](https://travis-ci.org/certbot/website.svg?branch=master)](https://travis-ci.org/certbot/website)
 
-## Getting Started
+## Development
 
-### Install
+### With Docker
+
+`docker-compose up`
```
Comment (wioux) [inline README.md]: The docker build fails if you don't have the _docs submodule cloned. Those steps (4/5) are now under "Without Docker" which is a little confusing.

File: all-instructions/index.html [hunk 2]
```diff
@@ -1,53 +0,0 @@
----
```
Comment (wioux) [inline all-instructions/index.html]: The docker build fails when it tries to copy this deleted directory https://github.com/certbot/website/blob/f1c9566c5b60f7e71d78c631fe19916f64a5b79d/Dockerfile#L53

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    image: nginx
+    ports:
+      - '4000:4000'
+    volumes:
+      - assets:/usr/share/nginx/html:ro
+      - ./nginx.conf:/etc/nginx/conf.d/default.conf
+  site:
+    build: .
+    volumes:
+      - .:/opt/certbot/
```
Comment (wioux) [inline docker-compose.yml]: This volume caused some issues for me. `npm install` / `_docs.sh depend/install` / `gulp build`  are run in the container as part of the Dockerfile, so my host directory did not initially have node_modules / build products.

File: _scripts/instruction-widget/main.js [hunk 4]
```diff
@@ -90,6 +81,15 @@ InstructionWidget = (function() {
     }
   };
 
+  // Users used to be able to link to an instruction set with an anchor link.
+  // We can redirect them to a standalone page.
+  redirect_anchor = function() {
+    var params = window.location.hash.replace('#', '').split('-');
+    if (params.length === 2) {
+      jump(params[0], params[1]);
```
Comment (wioux) [inline _scripts/instruction-widget/main.js]: This rewrote the url for me but didn't render any instructions.

http://localhost:4000/#debianjessie-nginx

File: all-instructions/index.html (hunk 2 above)
Comment (vbrown608) [reply all-instructions/index.html]: Thanks. I'll make a follow-up PR to handle this with docker-ignore. I didn't want to jam too much (more) into this one.

**Comment by wioux** [comment]
Everything looks good. :+1:

I think what jsha suggested about page titles in #261 is a good idea worth pulling out into a new issue.

**Comment by vbrown608** [comment]
Thanks @wioux, I'll merge this when changes to the nginx config are live.

**Pull request merged**